packages/realm/src/app-services/SyncSession.ts

```diff
@@ -93,6 +109,21 @@ function toBindingDirection(direction: ProgressDirection) {
   }
 }
 
+function isEstimateProgressNotification(cb: ProgressNotificationCallback): cb is DynamicProgressNotificationCallback {
+  return cb.length === 1;
+}
+
+function toBindingProgressNotificationCallback(cb: ProgressNotificationCallback) {
+  if (isEstimateProgressNotification(cb)) {
+    return (transferredBytes: binding.Int64, transferrableBytes: binding.Int64, progressEstimate: number) =>
+      cb(progressEstimate);
+  } else {
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    return (transferredBytes: binding.Int64, transferrableBytes: binding.Int64, _: number) =>
+      cb(transferrableBytes, transferrableBytes);
+  }
+}
```

Comment:
* Nit: Argument name
* The 2nd callback was accidentally called without `transferredBytes`.
```suggestion
function toBindingProgressNotificationCallback(callback: ProgressNotificationCallback) {
  if (isEstimateProgressNotification(callback)) {
    return (_: binding.Int64, __: binding.Int64, progressEstimate: number) => callback(progressEstimate);
  } else {
    return (transferredBytes: binding.Int64, transferrableBytes: binding.Int64, _: number) =>
      callback(transferredBytes, transferrableBytes);
  }
}
```